valhalla/mjolnir/node_expander.h

```diff
@@ -97,12 +97,13 @@ struct Edge {
       e.attributes.driveableforward = false;
       e.attributes.driveablereverse = false;
     } else {
-      e.attributes.driveableforward = way.auto_forward();
-      e.attributes.driveablereverse = way.auto_backward();
+      e.attributes.driveableforward = way.auto_forward() || way.truck_forward();
```

Comment:
This doesn't seem right. If we have motorcar or hgv only ways the || will mark them as usable by both for the purposes of reclassification but in the end its possible they are exclusive. That's why I said in my comment the fix will be tricky.